pkg/dbmate/migration_test.go

```diff
@@ -88,16 +101,30 @@ create table users (id serial, name text);
 ALTER TYPE colors ADD VALUE 'orange' AFTER 'red';
 -- migrate:down transaction:false
 ALTER TYPE colors ADD VALUE 'orange' AFTER 'red';
+-- migrate:up transaction:false
+ALTER TYPE colors ADD VALUE 'green' AFTER 'red';
+-- migrate:down transaction:false
+ALTER TYPE colors ADD VALUE 'green' AFTER 'red';
 `
 
-		parsed, err := parseMigrationContents(migration)
+		parsedSections, err := parseMigrationContents(migration)
+		parsedFirstSection := parsedSections[0]
+		require.Nil(t, err)
+
+		require.Equal(t, "-- migrate:up transaction:false\nALTER TYPE colors ADD VALUE 'orange' AFTER 'red';\n", parsedFirstSection.Up)
+		require.Equal(t, false, parsedFirstSection.UpOptions.Transaction())
+
+		require.Equal(t, "-- migrate:down transaction:false\nALTER TYPE colors ADD VALUE 'orange' AFTER 'red';\n", parsedFirstSection.Down)
+		require.Equal(t, false, parsedFirstSection.DownOptions.Transaction())
+
+		parsedSecondSection := parsedSections[1]
 		require.Nil(t, err)
 
-		require.Equal(t, "-- migrate:up transaction:false\nALTER TYPE colors ADD VALUE 'orange' AFTER 'red';\n", parsed.Up)
-		require.Equal(t, false, parsed.UpOptions.Transaction())
+		require.Equal(t, "-- migrate:up transaction:false\nALTER TYPE colors ADD VALUE 'green' AFTER 'red';\n", parsedSecondSection.Up)
+		require.Equal(t, false, parsedSecondSection.UpOptions.Transaction())
 
-		require.Equal(t, "-- migrate:down transaction:false\nALTER TYPE colors ADD VALUE 'orange' AFTER 'red';\n", parsed.Down)
-		require.Equal(t, false, parsed.DownOptions.Transaction())
+		require.Equal(t, "-- migrate:down transaction:false\nALTER TYPE colors ADD VALUE 'green' AFTER 'red';\n", parsedSecondSection.Down)
+		require.Equal(t, false, parsedSecondSection.DownOptions.Transaction())
 	})
```

Comment:
I think this test would be more meaningful if:

  * the second pair of up/down had `transaction:true`
  * a third pair of up/down have no options at all, but within the same migration file
  * a fourth and fifth pair have mixed options, the fourth having options on the up but not on the down, and the fifth having options on the down but not the up

I think this would fully cover all the possible scenarios, and also demonstrate the options are actually being parsed independently in each section and not carried over or reused from a previous section.